src/utils/operator-job.ts

```diff
@@ -85,33 +85,49 @@ export abstract class OperatorJobAwareCommand extends HealthCheck {
     operatorJobPayload: string,
     tags: (string | number)[],
   ): Promise<OperatorJob | undefined> {
-    const contract: Contract = this.networkMonitor.operatorContract.connect(this.networkMonitor.providers[network])
-    const rawJobDetails: any[] = await contract.getJobDetails(operatorJobHash)
-    const jobDetails: OperatorJobDetails = {
-      pod: rawJobDetails[0] as number,
-      blockTimes: rawJobDetails[1] as number,
-      operator: (rawJobDetails[2] as string).toLowerCase(),
-      startBlock: rawJobDetails[3] as number,
-      startTimestamp: BigNumber.from(rawJobDetails[4]),
-      fallbackOperators: rawJobDetails[5] as number[],
-    } as OperatorJobDetails
-    if (jobDetails.startBlock > 0) {
-      // for some reason these logs dont have tag attached.
+    try {
+      const contract: Contract = this.networkMonitor.operatorContract.connect(this.networkMonitor.providers[network])
+
+      // Try to fetch job details
+      const rawJobDetails: any[] = await contract.getJobDetails(operatorJobHash)
+
+      // Validate rawJobDetails before processing
+      if (!rawJobDetails || rawJobDetails.length < 6) {
+        throw new Error(`Invalid job details for job ${operatorJobHash}`)
+      }
+
+      const jobDetails: OperatorJobDetails = {
+        pod: rawJobDetails[0] as number,
+        blockTimes: rawJobDetails[1] as number,
+        operator: (rawJobDetails[2] as string).toLowerCase(),
+        startBlock: rawJobDetails[3] as number,
+        startTimestamp: BigNumber.from(rawJobDetails[4]),
+        fallbackOperators: rawJobDetails[5] as number[],
+      } as OperatorJobDetails
+
+      if (jobDetails.startBlock <= 0) {
+        throw new Error(`Invalid startBlock for job ${operatorJobHash}`)
+      }
 
       this.networkMonitor.structuredLog(network, `Decoded valid job ${operatorJobHash}`, tags)
       this.networkMonitor.structuredLog(network, `Selected operator for job is ${jobDetails.operator}`, tags)
+
       const targetTime: number = this.getTargetTime(network, jobDetails)
-      // extract gasLimit and gasPrice from payload
+
+      // Extract gasLimit and gasPrice from payload
       const gasLimit: BigNumber = BigNumber.from('0x' + operatorJobPayload.slice(-128, -64))
       this.networkMonitor.structuredLog(network, `Job gas limit is ${gasLimit.toNumber()}`, tags)
+
       const gasPrice: BigNumber = BigNumber.from('0x' + operatorJobPayload.slice(-64))
       this.networkMonitor.structuredLog(network, `Job maximum gas price is ${formatUnits(gasPrice, 'gwei')} GWEI`, tags)
+
       const remainingTime: number = Math.round((targetTime - Date.now()) / 1000)
       this.networkMonitor.structuredLog(
         network,
         `Job can be operated ${remainingTime <= 0 ? 'immediately' : 'in ' + remainingTime + ' seconds'}`,
         tags,
       )
+
```

Comment:
@alexanderattar So in the next line we add the next job to the `this.operatorJobs` object. This object gets consumed by another function that then executes the object by hash. I think we may want to log the size of the `this.operatorJobs` object. We can log the number of keys. This may help us understand if this array is getting too big.

Reply:
Good call. I can add it